### ISSUE TITLE
1.13+ Support?

### ISSUE THREAD

**Issue by 3ssej** [APDevTeam/Movecraft#63]
Hello,
I just found this particular updated GitHub repository for Movecraft, and I was hoping to use it on 1.13 but it doesn't seem to work at the moment.
Here is the error I've been getting (I've tried running it alone and clearing the configuration files but this error still persists)

`
16.08 09:58:36 [Server] INFO [Movecraft] Recognized PilotTool setting of: 280 16.08 09:58:36 [Server] WARN java.lang.ClassNotFoundException: net.countercraft.movecraft.compat.v1_13_R1.IWorldHandler 16.08 09:58:36 [Server] WARN at java.net.URLClassLoader.findClass(URLClassLoader.java:381) 16.08 09:58:36 [Server] WARN at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:154) 16.08 09:58:36 [Server] WARN at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:100) 16.08 09:58:36 [Server] WARN at java.lang.ClassLoader.loadClass(ClassLoader.java:424) 16.08 09:58:36 [Server] WARN at java.lang.ClassLoader.loadClass(ClassLoader.java:357) 16.08 09:58:36 [Server] WARN at java.lang.Class.forName0(Native Method) 16.08 09:58:36 [Server] WARN at java.lang.Class.forName(Class.java:264) 16.08 09:58:36 [Server] WARN at net.countercraft.movecraft.Movecraft.onEnable(Movecraft.java:155) 16.08 09:58:36 [Server] WARN at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) 16.08 09:58:36 [Server] WARN at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:339) 16.08 09:58:36 [Server] WARN at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:403) 16.08 09:58:36 [Server] WARN at org.bukkit.craftbukkit.v1_13_R1.CraftServer.enablePlugin(CraftServer.java:425) 16.08 09:58:36 [Server] WARN at org.bukkit.craftbukkit.v1_13_R1.CraftServer.enablePlugins(CraftServer.java:339) 16.08 09:58:36 [Server] WARN at net.minecraft.server.v1_13_R1.MinecraftServer.m(MinecraftServer.java:562) 16.08 09:58:36 [Server] WARN at net.minecraft.server.v1_13_R1.MinecraftServer.g_(MinecraftServer.java:524) 16.08 09:58:36 [Server] WARN at net.minecraft.server.v1_13_R1.MinecraftServer.a(MinecraftServer.java:423) 16.08 09:58:36 [Server] WARN at net.minecraft.server.v1_13_R1.DedicatedServer.init(DedicatedServer.java:288) 16.08 09:58:36 [Server] WARN at net.minecraft.server.v1_13_R1.MinecraftServer.run(MinecraftServer.java:686) 16.08 09:58:36 [Server] WARN at java.lang.Thread.run(Thread.java:748) 16.08 09:58:36 [Server] ERROR [Movecraft] Could not find support for this version.
`

My spigot version is `CraftBukkit version git-Spigot-417cfb3-0e1d79b (MC: 1.13) (Implementing API version 1.13-R0.1-SNAPSHOT)` which is a fair bit outdated so I will update my server and see if anything changes.

I would appreciate any help you can give me.

Thanks,
Jesse

**Comment by cccm5** [comment]
1.13 support is a bit touchy, as type and data have been completely removed from spigot. Previously there was several system that *required* using type and data to modify blocks safely. While those system now exist with the Material api in the 1.13, they do not exist in previous version in the same form. Typically, to add support for another version all that would be needed is a new WorldHandler, but I don't know if the same system will work on 1.13 due to the changes in how data is stored. I'll look into if it can be done, but there's a chance that it won't.

**Comment by cccm5** [comment]
Another issue is the addition of waterlogged blocks, which to support would require some refactoring. I'll probably hold off on adding 1.13 until after the initial release of this fork.

**Comment by 3ssej** [comment]
That sucks. Would you still be continuing Movecraft development in the far future even though you might not be able to support the latest version? Also, if you are able to add 1.13 support, when do you think that would be? A month? Sooner? Later? 
The reason I was hoping for 1.13 compatibility is because I've upgraded my world already (and made changes), and even if I loaded a backup in 1.12, newly developed plugins (which are mostly going to be based on 1.13 once it's entirely stable) will not be backwards compatible due to the way that 1.13 Spigot was developed.

Thanks for all of your help

**Comment by cccm5** [comment]
I'll still be supporting movecraft for the foreseeable future. Spigot on 1.13 isn't even in a stable position yet, so I probably won't update until than.

**Comment by FrozenBeard** [comment]
Started testing 1.13 today with my plugins, and cryed when there was no compatibility in Movecraft.
What do you need to get this fork out to 1.12 first? and then what do you need to start 1.13 work?

**Comment by keough99** [comment]
From what I've seen with the API for 1.13 it completely changed the entire
system used for blocks and material ids and does a lot with NBT stuff. It
would probably require a rewrite of the way Movecraft detects blocks

**Comment by FrozenBeard** [comment]
With the new data packs in 1.13 being able to create smooth moving blocks, and the 1.14 instant teleport, at some point in time Movecraft could be made 100% smooth.

**Comment by cccm5** [comment]
The changes 1.13 will most likely be an (almost) complete rewrite in preparation for the performance change in 1.14. I'm going to attempt to remove all uses of nms code from Movecraft so that its completely future compatible (baring any api regressions). There were two reasons nms was used previously - lighting updates and tile entity creation - but 1.14 is changing how lighting works and I expect some changes to be done around tile entities as well. 1.13 also introduces several changes itself that are better implemented without backwards compatibility - for example, water logged blocks and the flattening of blocks data.
On the two things you mentioned, the instant teleport, as far as I'm aware, was actually a client-side bug and is already possible (and implemented). I'm no sure if by smooth moving blocks you mean some shulker box magic, but some more details would be cool. 

A rewrite allows the following new possibilities
* Water logged ship blocks
* simple phasing through all blocks that can be passed through by a player
* Cleaner configs (No more need for block data due to flattening)
* Better async programming

I doubt a rewrite for 1.13 will take nearly as long as the current rewrite - I've gained a large amount of experience and most of the time for the current update was spent on fixing the previous code base to make it readable. 
That being said - I intend to only start work on 1.13 once the 1.10-1.12 version is complete

**Comment by FrozenBeard** [comment]
@cccm5 If you know what needs to happen and how, then its just motivation! 1.13 is fine to have a re-write most plugins seem to have 1.7-1.12.2 versions and then a 1.13+ version of their plugin, For example FAWE does this and manages projects with github releases.
There is no way on earth the "official" movecraft is going to be updated at any point, This project would absolutely explode if it was updated. At PirateCraft we are reliant on Movecraft, the server resolves around it, so i'm stuck on 1.12.2 until I find an alternative (LOL) or this gets a sweet update.

So your 1.10-1.12.2 version, is there a list of what needs to be done? I think I ended up doing the Majority of testing with my users on it! we stopped when I posted all tickets for things we found and are eagerly awaiting a release for next set of testing.

The smooth blocks of 1.13 was in relation to the data packs addition;
https://www.youtube.com/watch?v=O9Fu7FRYWuI
https://www.youtube.com/watch?v=F0DExDqtZWs

How can we get this project up and running!

**Comment by keough99** [comment]
There is a way to support multiple different versions in 1 jar. I’ve actually seen universal jars done where 1 jar supports multiple different server softwares and it would be the same for different versions. My navycraft plugin runs a piece of code when the server enables the plugin to detect what version the server is running and enables the proper bits of code for that version. But if using the universal jar method we can do the same but have different minecraft versions with vast api changes in 1 jar file. So supporting 1.13 along with other versions of minecraft in the same jar is possible

**Comment by cccm5** [comment]
Yes, I;m aware that there's a way to have multiple versions in one jar - this plugin makes use of it already - but the 1.13 goal is to become compatible with all versions without the need for version specific support.

**Comment by hmmcrunchy** [comment]
Hi cccm thanks for the update 

We have a server which is pretty much dependant on Movecraft. Is it worth at present remaining on 1.12 if there isn’t a near date for Movecraft update ? Then perhaps looking to update later

**Comment by DerPavlov** [comment]
Was there already some progress for 1.13? Otherwise I would perform the flattening.

**Comment by cccm5** [comment]
At the moment I have not started any work towards 1.13, however I know @eirikh1996 is looking into it on his own.

**Comment by FrozenBeard** [comment]
Itching for a new release! My main holdup to move to 1.13! Can't have a ship server without ships 😂

**Comment by DerPavlov** [comment]
Let me know when you need support (e.g.: copy paste the configs to the new flattend names)

**Comment by FrozenBeard** [comment]
Where are we at with 1.13?

**Comment by DerPavlov** [comment]
I made a test-plugin with Worldedit/FastAyncWorldEdit to move structures. Sadly, I didn't have the time to finish the performance tests.

**Comment by cccm5** [comment]
as far as I know this project moved away from worldedit/async-worldedit due to it chunking the movement of ships as well as not handling the movement of tiles that are currently being processed (i.e. redstone clocks). It will work, but clients will see visual breaking of structures and also loose the ability to use most redstone devices

**Comment by lordegeorge** [comment]
Is this plugin still being updated, and if so, will 1.13.2 support come soon? Or even 1.14?

**Comment by cccm5** [comment]
there are independent efforts from a few developers to add support for 1.13, but they're also waiting on the main branch to have some major bug fixes.

**Comment by lordegeorge** [comment]
Alright. I am 3ssej by the way.

**Comment by FrozenBeard** [comment]
I too am still very interested in updated version

**Comment by XGNXLency** [comment]
My server developer for Galactic Craft is working on a fork of Movecraft for 1.13.2, which will be intended to serve as a replacement for the original Movecraft plugin for 1.12.2 and below. So far he's reached out to some of the original team, but they've been uninterested with reviving the original plugin. They have preferred to make an entirely new plugin, the details of which I'm not exactly knowledgeable to. All I know is that very few people have expressed interest in reviving the old plugin for 1.13.2. He's also reached out to a server called Navy Craft (which isn't that good, but I'll save that for a later debate) that unfortunately has the most progress as I know. They have absolutely zero interest in releasing a Movecraft plugin to the public and are continuing to develop it for their own server. I think it would be in everyone's best interest if we all come together, make a Movecraft plugin with the intent of actually adding modifications to it once a stable foundry is produced. At the moment, all we need is to work out the critical bugs and mechanics, and then we can all make forks for our servers with our private developing system.

**Comment by cccm5** [comment]
I mean it's against the movecraft licence to maintain a private fork, but I have no intention of going after people for it. If they make a new plugin from scratch however, that's perfectly legally fine. Most of the original movecraft developers are no longer working on the project (it's gone through a few mantainers), and this is now the official current repo. I'm always open to pull requests.

**Comment by XGNXLency** [comment]
> 
> 
> I mean it's against the movecraft licence to maintain a private fork, but I have no intention of going after people for it. If they make a new plugin from scratch however, that's perfectly legally fine. Most of the original movecraft developers are no longer working on the project (it's gone through a few mantainers), and this is now the official current repo. I'm always open to pull requests.

So far, I'm trying to get a few devs to work together to completely remake the original Movecraft for 1.13.2. Afterwards, private forks can be made. Legal issues shouldn't apply to those who helped contribute to the plugin's creation, correct?

**Comment by DerPavlov** [comment]
I wrote as a test a Worldedit based craft plugin, just to verify if it is feasible. 
https://github.com/DerPavlov/Ironclad

The good news, it works with Worldedit (200ms for a 5000 Block ship) but even better with AsyncWorldEdit (30ms for a 5000 Block ship). But ship controls and damage system are still missing though.

Unfortunately, I am very busy with a different project and don't have much time to finish the plugin.

**Comment by XGNXLency** [comment]
> 
> 
> I wrote as a test a Worldedit based craft plugin, just to verify if it is feasible.
> https://github.com/DerPavlov/Ironclad
> 
> The good news, it works with Worldedit (200ms for a 5000 Block ship) but even better with AsyncWorldEdit (30ms for a 5000 Block ship). But ship controls and damage system are still missing though.
> 
> Unfortunately, I am very busy with a different project and don't have much time to finish the plugin.

If you'd like, we can get you on our team and we can assist with making a public Movecraft. Our goal is to completely recreate the original Movecraft for 1.13.2.

**Comment by discapes** [comment]
> > I wrote as a test a Worldedit based craft plugin, just to verify if it is feasible.
> > https://github.com/DerPavlov/Ironclad
> > The good news, it works with Worldedit (200ms for a 5000 Block ship) but even better with AsyncWorldEdit (30ms for a 5000 Block ship). But ship controls and damage system are still missing though.
> > Unfortunately, I am very busy with a different project and don't have much time to finish the plugin.
> 
> If you'd like, we can get you on our team and we can assist with making a public Movecraft. Our goal is to completely recreate the original Movecraft for 1.13.2.

Whats wrong with the current one?

**Comment by Pulverizer** [comment]
> > I mean it's against the movecraft licence to maintain a private fork, but I have no intention of going after people for it. If they make a new plugin from scratch however, that's perfectly legally fine. Most of the original movecraft developers are no longer working on the project (it's gone through a few mantainers), and this is now the official current repo. I'm always open to pull requests.
> 
> So far, I'm trying to get a few devs to work together to completely remake the original Movecraft for 1.13.2. Afterwards, private forks can be made. Legal issues shouldn't apply to those who helped contribute to the plugin's creation, correct?

Wrong. Any use of the code that has been added under the license in use has to use that license unless you wrote the code. If you wrote that section of code then you should be fine?

**Comment by Pulverizer** [comment]
> I wrote as a test a Worldedit based craft plugin, just to verify if it is feasible.
> https://github.com/DerPavlov/Ironclad
> 
> The good news, it works with Worldedit (200ms for a 5000 Block ship) but even better with AsyncWorldEdit (30ms for a 5000 Block ship). But ship controls and damage system are still missing though.
> 
> Unfortunately, I am very busy with a different project and don't have much time to finish the plugin.

Even with AsyncWorldEdit that's still 2-3x slower than what we have atm IIRC.

**Comment by cccm5** [comment]
The issue with using async worldedit is that it splits the movement into parts, which causes redstone mechanisms to breaks. Also, as pulverizer said, that's still slower than what we have now.

**Comment by DerPavlov** [comment]
It might be slower or faster, who knows. You would need a crystal ball to answer this question. So far there is no 1.13 version of Movecraft I could use for benchmarking. I believe, a working version for 1.13 might be the step in the right direction and many people might use it, even if it is slower. And the code for moving the craft is just a very small section, which can be replaced by a more optimized code anytime. 
So far I didn't see any progress for Movecraft 1.13, the version of Minecraft was released almost a year ago and 1.14 is coming soon. If you believe 1.12 is the version to go, it is good for me.

**Comment by wolfebersahd** [comment]
I kinda have to agree with DerPavlov, Movecraft was/is the premier ships plugin publicly available. I would gladly pay for a premium version that is updated quickly and is optimized for specific server versions.

I understand there is a huge change to move from 1.12.2 to 1.13.X, then probably a slightly easier jump to 1.14.X.

I know myself and @FrozenBeard are eagerly awaiting the 1.13.X version of Movecraft because it's one of the only plugins (in some cases maybe the only one) left to be updated to 1.13.X.

If the DEV team is using 1.12.X as a learning tool to better optimize the 1.13, and future versions, then I am all for it.

But, and I say this with a with utmost respect for you guys, and gratitude for taking over this project and maintaining it, I do find it is taking a long time.

If there is anything the non-coder community can do to expedite your development process, please let us know.

**Comment by XGNXLency** [comment]
Okay. So this is very clear as is. There are a lot of legal issues and problems with each of our own forks of Movecraft. My point is, it doesn't help doing a whole bunch of different private forks rather than coming together and contributing to a remake of the 1.12.2 version for 1.13.2. The more people we have working together, the easier it is to finish it quickly and efficiently. I'm 100% the main goal at the moment is to completely remake the 1.12.2 version exactly as it is for 1.13.2, maybe with a few improvements or new features. At the moment, we should all just wait to branch off and make our own private forks until we've got an actual public release working.

**Comment by discapes** [comment]
Eirikh has a working 1.13.x fork i believe

**Comment by TylerS1066** [comment]
> a remake of the 1.12.2 version for 1.13.2

I think you all misunderstand the status of Movecraft.  The 1.12 version is nowhere near stable and has many core bugs left to fix, of which the [milestone found here](https://github.com/APDevTeam/Movecraft/milestone/2) lists (atleast as of a month or so ago).  

We first need to get the core plugin working on 1.12, then we can work toward 1.13 and later versions.

**Comment by XGNXLency** [comment]
> 
> 
> > a remake of the 1.12.2 version for 1.13.2
> 
> I think you all misunderstand the status of Movecraft. The 1.12 version is nowhere near stable and has many core bugs left to fix, of which the [milestone found here](https://github.com/APDevTeam/Movecraft/milestone/2) lists (atleast as of a month or so ago).
> 
> We first need to get the core plugin working on 1.12, then we can work toward 1.13 and later versions.

Okay but why? The 1.12.2 plugin we used to use worked fine. It's public on PMC and everything. 1.13.2 should be the bigger concern because a public working release doesn't even exist for it.

**Comment by cccm5** [comment]
the version that "worked fine" had chances to cause corruption and had an unreadable codebase. This project is designed to work on *mutiple* versions using the same jar, but there's no point in trying to add support for another version if the non version specific code is also not working.

**Comment by TylerS1066** [comment]
> chances to cause corruption

Let me clarify some of the issues in movecraft 6.0:
1. Commonly causes corruption
2. Assaults completely break, possibly destroying airspaces
3. Sieges do not work
4. Repairs are very buggy
5. Lag is unbearable in large tile heavy worlds
6. Randomly the entire plugin will crash (we had this happen on AP this weekend during a major battle, screwing up everything).
7. Explosion resistance makes blocks invulnerable to torpedo type crafts.
8. And many more I'm forgetting.


If you want to help get the core of 7.0 ready for a full release please look at the list of open issues (specifically ones I highlighted in that milestone) and assign one to yourself!   The fastest way for all of us to get a stable 1.12 base (and a 1.13 ready version) is to fix those bugs.

As to 1.13 once we get a 1.12 version, I believe eirik has some good progress that we could easily start implementing ONCE we get a stable 1.12 version.

**Comment by cccm5** [comment]
They're referring to movecraft 5 I believe, but I believe similar things apply

**Comment by TylerS1066** [comment]
> They're referring to movecraft 5 I believe, but I believe similar things apply

*shudders at worse bugs*

**Comment by XGNXLency** [comment]
Very well. If we all work together, I bet there will be progress increasing. I've already hosted a Discord channel involving 2 devs working on a 1.13.2 release. If you want an invite, I can post it here.

**Comment by discapes** [comment]
We already have a discord server

**Comment by XGNXLency** [comment]
> 
> 
> We already have a discord server

Then do tell what results have been produced.

**Comment by discapes** [comment]
bit.ly/joinAP-Dev

**Comment by XGNXLency** [comment]
![2019-04-16 (4)](https://user-images.githubusercontent.com/21114145/56190675-94c82080-5fdf-11e9-9fb9-5f44569afe18.png)
Super.

**Comment by TylerS1066** [comment]
http://bit.ly/JoinAP-Dev

**Comment by SirEraz** [comment]
I'm not sure if anyone has already seen this, but I recently came across a fork of Movecraft by C_Corp2002/C_Corp2k2. He says that his fork works on Spigot 1.13.2, although I personally haven't been able to test anything yet.
Here's a Gitlab link: [](https://gitlab.com/ccorp2002/movecraft-8.x)
And also his PMC post: [](https://www.planetminecraft.com/mod/movecraft-1-10-2-gt-1-13-2-spigot-bukkit-c-corp2002-s-version/)

**Comment by cccm5** [comment]
There is no code in that repo aside from the compiled Jar, of which he just modified this repo and called it "movecraft 8." ccorp has in the past already violated the license of this plugin, I'm not surprised he's done it again. To make matters worse, he didn't even give attribution to any people that worked on it after baccayarro *again* - seeming to claim all the work as his own.

**Comment by SirEraz** [comment]
@cccm5 Really? That's ridiculous - he didn't even make changes? That's really upsetting.

**Comment by cccm5** [comment]
Naw, he did add some code for 1.13 supourt and some repair mode changes - I've talked it out with him and he's going to go through the proper route now. Thanks for bringing it to my attention @SirEraz

**Comment by SirEraz** [comment]
Oh, ok, that's great to hear. Glad I could help! @cccm5

**Comment by XGNXLency** [comment]
Known bugs on 1.13.2:
+ The harvestBlocks function does not work in the .craft files.
+ Ships have a tendancy to split apart when using Ascend or Descend.
+ There are no /rotateleft and /rotateright commands.
+ The lighting on ships tends to break or cover the ship in darkness.
+ The localization has some missing messages, such as the message displayed when turning too quickly.
+ There's no option to disable or enable the GUI messages when turning Cruise on and off.
+ There's no config to edit what the GUI messages say when using Cruise.
+ The Helm Sign is case sensitive, and only works when the user types in [helm].
+ Ships sometimes leave behind cloned parts or large chunks from the parent ship.
+ Directional blocks with inventories such as furnaces and dispensers have a tendancy to crash players.
Error Message: "The game crashed whilst rendering block entity
Error: java.lang.IllegalArgumentException: Cannot get property bme{name=facing, clazz=class eq, values=[north, south, west, east]} as it does not exist in Block{minecraft:dispenser}"
+ Ships have a tendancy to break apart during a server restart, even when they're released.

**Comment by themindstorm** [comment]
I haven't tested this yet, but would **[ViaVersion](https://www.spigotmc.org/resources/viaversion.19254/)** work? The server version can be 1.12, while players can be on 1.12, 1.13, or even 1.14

**Comment by Pulverizer** [comment]
That's what the official server is using right now for the current version of Movecraft.

**Comment by TylerS1066** [comment]
> I haven't tested this yet, but would **[ViaVersion](https://www.spigotmc.org/resources/viaversion.19254/)** work? The server version can be 1.12, while players can be on 1.12, 1.13, or even 1.14

Correct, Viaversion works fine for allowing newer clients (and what Airship Pirates) currently is using.  I believe everyone wants native 1.13 support, not just "faked" support with viaversion.

**Comment by FrozenBeard** [comment]
PirateCraft is having to use ViaVersion for now too.

**Comment by cccm5** [comment]
As an update for you all, I'm working on this in the kotlin rewrite branch

**Comment by joelemail** [comment]
I'm wondering if there is currently support for 1.14.  It looked like Dan (cccm5) was working on it, but that was in May, perhaps?

**Comment by cccm5** [comment]
I'm currently aiming to get the full release for 1.12 out at which point I'm going to fully transition to new version compat most likley.

**Comment by XLentGamingNetwork** [comment]
Jesus, you guys are still working on the 1.12 update. Guys, I'd just drop it and focus on 1.13 and 1.14. 1.12 is nearly dead by this point. By the time you guys are finished the next big update will come out with its whole new set of problems. Priorities are self-evident at this point.

**Comment by cccm5** [comment]
> 
> 
> Jesus, you guys are still working on the 1.12 update. Guys, I'd just drop it and focus on 1.13 and 1.14. 1.12 is nearly dead by this point. By the time you guys are finished the next big update will come out with its whole new set of problems. Priorities are self-evident at this point.

A bug free release is the main priority. I'm not adding 1.13 support as a priority over fixing bugs that can have a negative impact on user's worlds.

**Comment by ccorp2002** [comment]
Ahh yes.. you again. @XLentGamingNetwork --- or should I call you GalacticCraft, Ren, or XGNLency
Yes i just pinged you... No i am not looking for a fight.

**Comment by TheRealBotcrusher** [comment]
Yes, the priorities are self evident. Movecrraft is a complicated plugin with lots of potential for things to break due to the Minecraft "engine" 1.13 brings a whole other level of headaches due to changes in handling block manipulation as mentioned at the top of this chain.

**Comment by XLentGamingNetwork** [comment]
> 
> 
> Ahh yes.. you again. @XLentGamingNetwork --- or should I call you GalacticCraft, Ren, or XGNLency
> Yes i just pinged you... No i am not looking for a fight. Dont you think i have forgotten.

C_Corp, don't even get me started on your bullshittery. You steal ccm's code, stab me in the back TWICE when I offer to help you, and then proceed to TRASH TALK ME behind my back. Then you have the audacity to accuse me of DDoSing your server when I've never done so to anyone IN MY LIFE. The ONLY reason I haven't started making posts against you EVERYWHERE I've gone is because I have ALLOWED YOU to keep whatever little respect you have left. So don't try me.

**Comment by XLentGamingNetwork** [comment]
> 
> 
> Yes, the priorities are self evident. Movecrraft is a complicated plugin with lots of potential for things to break due to the Minecraft "engine" 1.13 brings a whole other level of headaches due to changes in handling block manipulation as mentioned at the top of this chain.

I understand that, which is why it would be best to just abandon support for 1.12 temporarily and get the 1.13/1.14 version working first. 1.12 was a disaster as is, and frankly 1.13 and 1.14 are probably more manageable to deal with than 1.12 ever was. Plus, everyone is demanding 1.13/1.14 support anyways. I hardly see anyone asking for 1.12 support.

**Comment by TheRealBotcrusher** [comment]
You've missed Cccm's point, there is code that isn't version dependent that needs to be fixed before moving to 1.13 can be attempted. Trying to port bugged software without fixing the bugs wouldn't be an effective use of time.

**Comment by ccorp2002** [comment]
I may have stole his code... But at least I stopped when I was told, and unlike you; I feel guilt.
And trash talking?? No. I was warning others of you. And what help have you offered, besides the "partnership". And when you don't get your way, it appears to me, you go apeshit. Look. I don't trust you, and nor should anyone else.

**Comment by cccm5** [comment]
@ccorp2002 @XLentGamingNetwork  Please keep this thread on topic. This is not a place for you to argue.

**Comment by ccorp2002** [comment]
Fair point.

**Comment by XLentGamingNetwork** [comment]
Then he should not be attracting unwanted attention by encouraging it. You want to point a finger, point it at him.

**Comment by ccorp2002** [comment]
Well... Your the one perpetuating it by heading back into this. No, I don't want any private developers for movecraft I'm just staying here so YOU don't come in and rip movecraft out of their hands with some underhanded trick.

**Comment by ccorp2002** [comment]
I know your type. You will stop at nothing to get what you want. What you "deserves". For all I know, you are a 13-15 year old nitwit based on your YouTube videos. Yes I watched them to learn more information.

**Comment by cccm5** [comment]
@XLentGamingNetwork @ccorp2002 if *either* of you continue I will ban you from participating in this repo.

**Comment by ccorp2002** [comment]
Okay.

**Comment by Mysteri02** [comment]
found this on PMC thought yall should see it. I tend to agree with the poor lad and steps should be taken to help him. Had it in my cache but the thread was deleted so ill copy and paste it here

**This is a brief summary and a warning. If you want full details, please proceed below this paragraph.** Beware of this user!: C_Corp2002#8261 He's been stealing code from developers of different games, one such game being Minecraft. He's also been trash talking people behind their backs and turning people against each other. If you see him, block him and ban him from your Discord and server if you see him. Spread the word please. A link to his Planet Minecraft profile can be found at the bottom of this thread along with links to the Movecraft GitHub, mod page, and C_Corp's GitHub profile.

	
**Before I start, I would just like to point out that this thread is meant to be taken seriously, despite this being a problem with a game or something that others would view as trivial. Despite this being a game, these events took place with REAL PEOPLE and had REAL CONSEQUENCES. If you are not willing to take this seriously then this is not the thread for you.**

At about the beginning of this year, I noticed there was a severe drop in popularity and quantity of Movecraft servers. After I had accidentally forced my server to update to 1.13 and corrupted all of the world files, I started looking around for a Movecraft plugin for 1.13. Since the latest Movecraft had ever been updated to was 1.12, I started seeking out help for an update. One of the unfortunate people I met was C_Corp2002. He claimed to have updated to 1.13 all by himself, yet in actuality he stole the code from the GitHub of cccm5's code and gave no credit. Before I knew he had stolen the code, all I did was find his server and comment on it offering my assistance if he ever needed help with configuring Movecraft or needing feedback. After I was nothing but polite, this user proceeded to threaten me and my server. Apparently he had heard some less than reputable comments about me from one of his "friends," who might I mention threatened to kill me and my girlfriend in real life. If you want proof, here's the Discord message from the user (JOCBA00). Please note that he had threatened myself and my girlfriend at two separate times. I'll just be covering the time he threatened me.

[https://i.imgur.com/Lqb3Rgx.png](url)

JOCBA00 was a banned player from my server. Unfortunately he also happened to play on C_Corp's server. When I tried to join C_Corp's server to explain the situation, I found out I was banned for no reason, having not previously joined at all. As you can already tell, this guy really doesn't like me. I joined the server on one of my alternate accounts and talked to one of his staff. That got C_Corps attention and I finally explained the situation. When me and my girlfriend joined his server for the first time (after I found out I was banned) we had joined at the time his server was apparently experiencing a DDoS attack. Later when I finally had a talk with C_Corp, he accused ME of DDoSing his server. For the record, I've never DDoSed anyone in my LIFE. I had to sit through a DDoS attack for nearly 6 months, so there's no way on the face of the planet that I'd want to DDoS anyone else. Regardless of how much I might hate someone for doing something, I would never resort to that. His "evidence" was claiming he saw the DDoS attack coming from my IP. I think he's just confusing the IP address message of every player that joins. If you're a server owner and you know how to use the console, it should be common knowledge than when ANY player joins, you're able to see his or her IP address. In any case, I'm getting off track.

Later on I found the REAL developer team of Movecraft lead by cccm5 on GitHub. cccm5 was the one who told me that C_Corp had actually stolen the code. cccm5 is however more focused on fixing the version for 1.12 rather than updating it to future versions. Before I met cccm5 and his team, I offered a partnership with C_Corp between my server's developer and his team. Since Movecraft was dying, I offered to make an organization that was meant for PR with Movecraft.

The goal was to have a neutral ground for all developers to contribute to updating Movecraft while at the same time establishing what the development process was going to be like. If anyone outside the development team had a question or a suggestion, the organization would relay the question to the developer team and give a response accordingly. The organization would also be responsible for overseeing the development of the project. The developers would be doing the coding and all of the hard stuff while the organization would just be keeping everyone on track. We're getting off topic again, so we'll leave that as it is. Eventually lots of progress was made and I even went as far as to defend C_Corp on the public GitHub page and cccm5's Discord server known as the APDevTeam. C_Corp claimed he didn't know the project was still being worked on and he didn't even bother giving credit anyways. What he told me was a lie.

Shortly after these events, my server started dying because C_Corp decided to stick his nose where it didn't belong. He started trash talking me behind my back with some of my staff members and convinced them to quit my server. He even managed to convince two of them that I was responsible for DDoSing his server because it happened again shortly after they quit. I went as far as even PROVING that I wasn't responsible because it was a Friday morning in June this year at around 10 in the morning. I was at school when the DDoS attack supposedly occurred. After I told them this, they still didn't believe me. I took a VIDEO of my current surroundings and showed them the classroom. The only thing I had with me was my laptop that's 4 years or so old and didn't even have an internet connection. They still didn't believe me. After that, I quit Minecraft thinking that it would just blow over and I wouldn't have to deal with it anymore. I shut down my server, shut down the organization, and stopped being involved with the development process altogether.

During this time I also developed a distaste for Minecraft altogether, and especially the APDevTeam. In my personal view even to this day, they don't have their priorities straight. You might say the same thing about me with this thread, but I'm doing this because I'm sick and tired of people succumbing to biased opinions such as the ones C_Corp has about me. I took the time to make this thread because I don't deserve being treated with disrespect for things I haven't even done.

A few days ago, I came back on the GitHub of Movecraft (which I'll post a link to below this thread) to see that they were STILL working on the 1.12 development. I should point out that they've been working on this for about two or so years, which is honestly a little ridiculous in my opinion. Here is a direct quote of what I posted on the GitHub.

_"Jesus, you guys are still working on the 1.12 update. Guys, I'd just drop it and focus on 1.13 and 1.14. 1.12 is nearly dead by this point. By the time you guys are finished the next big update will come out with its whole new set of problems. Priorities are self-evident at this point."_

Probably not the most subtle way to address the problems, but considering it's been going on for nearly two years I was sure I'd made my point. And as fast as a copyright claim takes down a YouTube video, guess who shows up? The good lord himself, C_Corp. Here's HIS direct quote to me.

_"Ahh yes.. you again. @XLentGamingNetwork --- or should I call you GalacticCraft, Ren, or XGNLency

Yes i just pinged you... No i am not looking for a fight. Dont you think i have forgotten."_

Frankly, he was looking for a fight. Despite what he says in that comment, that is a clear attempt at instigating hostile action. So I gave him what he wanted. Later on he went to say this.

_"I may have stole his code... But at least I stopped when I was told, and unlike you; I feel guilt. No I am not saying I am perfect. But at least I have humanity.
And trash talking?? No. I was warning others of you. And what help have you offered, besides the "partnership". And when you don't get your way, it appears to me, you go apeshit. Look. I don't trust you, and nor should anyone else."_

He was "warning others of me." It's honestly almost a laughing matter with this guy. He even ADMITTED stealing his code and felt "guilt" about it. If he felt guilty he wouldn't be doing what he is now. Keep in mind that this thread was made literally two days after he comments all of this on the GitHub. I offered help quite a bit as I said earlier. I offered to help configure Movecraft, make suggestions, and even be a neutral party so people could discuss ideas. I didn't get my way and I certainly didn't go "apeshit." As a matter a fact, I tried to just let the situation go and not look back. A couple of weeks ago I brought my server back up, but it's not open to the public. I wanted to do things differently to try and avoid clusterfucks like these. Eventually he goes on to make more hypocritical comments and prove my point that he feels absolutely no guilt about what he did. Here's some more of his comments.

_"Well... Your the one perpetuating it by heading back into this. No, I don't want any private developers for movecraft I'm just staying here so YOU don't come in and rip movecraft out of their hands with some underhanded trick."_

In what way was I going to "come in and rip Movecraft out of their hands with some underhanded trick?" Where in any of my comments did I say I was going to steal the code? It almost made me laugh at this guy's behavior. First off, he discredits code that is open source anyways, and then he accuses ME of trying to do the thing that he ALREADY DID. This is absolutely astonishing. He even goes on to GENERALIZE ME with a certain "type" and make biased assumptions like he did from the beginning.

_"I know your type. You will stop at nothing to get what you want. What you "deserve". For all I know, you are a 13-15 year old nitwit based on your YouTube videos. Yes I watched them to learn more information."_

This whole ordeal was beautiful. Just beautiful. More importantly, I'm "a 13-15 year old nitwit based on my YouTube videos." My YOUTUBE VIDEOS? You're going to judge me off of THOSE? Dude, come on. I do YouTube for FUN and ENTERTAINMENT. In NO WAY do I have some sort of grand scheme or master plan that's trying to undermine everyone's efforts. So in any case, I'm well aware that this thread may be deleted, reported, and marked as spam but at least I got to say my part. I just needed a place to rant. In all honesty, I would appreciate comments about this. I'm nearly 100% certain I'm not in the wrong here. I'm just a guy trying to live my life and finish some projects. Thank you for taking the time to read this.

Link to C_Corp's GitHub: https://github.com/ccorp2002

Link to C_Corp's Planet Minecraft Account: https://www.planetminecraft.com/member/c_corp2002/

Link to the Movecraft GitHub: github.com/APDevTeam/Movecraft/issues/63

Link to the Movecraft Mod Page: www.planetminecraft.com/mod/movecraft-reloaded---loraxe42s-fork/

**Comment by TheRealBotcrusher** [comment]
Trying to post with a thinly veiled alternate account you literally just made four hours ago with the sole intent of flaming ccorp is really the best way to advance the status of this issue....

This github is not the place to fan the flames of your feud.

**Comment by Mysteri02** [comment]
> 
> 
> Trying to post with a thinly veiled alternate account you literally just made four hours ago with the sole intent of flaming ccorp is really the best way to advance the status of this issue....
> 
> This github is not the place to fan the flames of your feud.
> 
> Once again, "dropping development" of the 1.12 version doesn't get you a working 1.13 version any faster, it gets you a broken version of movecraft for 1.13. Taking shots at the dev team that isn't being paid to do this gets you nowhere.

At this point, I'm just tired of being SHAT on whenever I make a SIMPLE COMMENT. I thought making a suggestion wouldn't get me ANY problems, but NOOOOO. SOMEONE had to come in and make the situation WORSE. Also, I'm not flaming him, I'm setting the record straight. I'm sick and tired of people giving me a bad reputation for things I didn't even do and things that people are saying I WANT to do. I'm not trying to do anything other than stop people from being jackasses to me because of biased opinions and rumors. I'm sick of it.

**Comment by Mysteri02** [comment]
If you want, you can mark the post as off topic. I'm not here to do anything OTHER than set the record straight, make suggestions, view progress, and maybe help out if requested. The only one here I have ANY hostile relation toward is C_Corp, and the post that's nearly 2,000 words explains why if you took the time to read it. Reading it is completely optional and you're free to come to your own conclusions. Just don't harass or make other people suffer because of your own personal grudges.

**Comment by hmmcrunchy** [comment]
Could we ban any further accounts posting nonsense as we don’t need to hear this

**Comment by Mysteri02** [comment]
> 
> 
> Could we ban any further accounts posting nonsense as we don’t need to hear this

Again, you can just mark the comments as OFF TOPIC as I've said already. It's hardly nonsense considering I'm not the one at fault here. I'm perfectly within my rights to try and prove that I'm not at fault.

**Comment by Mysteri02** [comment]
So unless someone else tries to take a shot at me again, I've said all I needed to. Again, you can arrive at whatever conclusion you want, but that thread is the truth by itself in near chronological order. I rest my case. Now can I PLEASE just go back on with my life and act like a normal person in this thread without having to defend myself at every corner?

**Comment by cccm5** [comment]
> 
> 
> Could we ban any further accounts posting nonsense as we don’t need to hear this

This is what is going to happen unfortunately. Anyone who continues talking about that topic will receive a ban. I don't want that to happen, so please don't continue.

**Comment by Mysteri02** [comment]
Okay good, since we've established that already, let me get back to what I was saying. I understand why you want to complete the 1.12 version, but so far I've hardly seen people asking about updating the 1.12 version. The versions everyone is wanting are for 1.13 and 1.14. Someone pointed out previously that if you stop supporting Movecraft for 1.12 and go to 1.13, all you'll get is a broken Movecraft 1.13 and an unfinished 1.12. I mean yeah, that's the case for any software on any game when updates are involved. That's part of the process. The trick is updating it fast enough to the latest version of the game. Since the transition from 1.12 to 1.13 was a major annoyance for almost every plugin out there, that alone is support on why 1.12 should be dropped and 1.13 should be supported. 1.12 works, I've hardly seen problems with it. I've not noticed nearly as many bugs with 1.12 as I have 1.13. The only things I've seen broken with 1.12 are the sieges and the gravity system, but that's it. Everything else was fine. I'm not saying drop 1.12 support altogether, but at least focus on getting the 1.13 version WORKING before we go back to support older versions.

**Comment by wolfebersahd** [comment]
Yeah, I really think this is more like 3 plugins in one with all the features.

I have to agree with hmmchrunchy that it probably would be better as 3 separate plugin.

They could all be part of the same plugin suite and the add-ons could even be premium to further support the future development of movecraft

**Comment by FrozenBeard** [comment]
Ive never used the siege feature as Movecraft never interacted with a proper Clans system like SimpleClans, if it did it would be super awesome, separating may increase speed to get it up and running faster, 1.15 is around the corner now!

Whats the biggest issues at present? I have a good community to test water based ships with pretty quick feedback, if I can help in any way to speed up getting movecraft updated please let me know!

**Comment by cccm5** [comment]
The issues with version compatibility are solley due to moving blocks, all the other "extra" stuff is version agnostic. Erik has already begun porting to 1.14/1.13, and there's several pull requests that fix most of the major bugs remaining for the current version (The biggest of them being the water hole issue).

**Comment by FrozenBeard** [comment]
Awesome that's good news, my biggest issue I couldn't figure out was people just getting kicked off the back of ships. Never figured that one out.

**Comment by wolfebersahd** [comment]
@FrozenBeard On the latest version it happened to me TWICE, with zero way of reproducing it. One was because of latency that was client side. Another was purely coincidental. I updated the closed bug with the issue #65

**Comment by StarrKiss** [comment]
Any update on this? As soon as I start the server I get a Java error w/ this and it shuts down becaue it checks for version

**Comment by cccm5** [comment]
> Any update on this? As soon as I start the server I get a Java error w/ this and it shuts down becaue it checks for version

The current version is still 1.10-1.12 only. @eirikh1996 is currently working on 1.13/1.14 support.

**Comment by VaasKahnGrim** [comment]
is there any updated info on a 1.15.2/1.16 version in the future yet?

**Comment by teamoneill4** [comment]
ive played on servers that have movecraft 1.15.2 how come it isnt updated on here

**Comment by memeries** [comment]
Hello, I'd like to ask, is there an ETA for future updates like 1.16? Totally understand if there's no ETA, because it's never good if updates are rushed. I was wondering if I could play this on a 1.16 spigot server with friends. Sorry if I'm asking for too much.

**Comment by TylerS1066** [comment]
> Hello, I'd like to ask, is there an ETA for future updates like 1.16? Totally understand if there's no ETA, because it's never good if updates are rushed. I was wondering if I could play this on a 1.16 spigot server with friends. Sorry if I'm asking for too much.


> I recommend heading over to [eirik's fork](https://github.com/eirikh1996/Movecraft) for experimental 1.13+ support.
> 
> We someday may be pulling the changes into this repository at such a time as it stabilizes, but we currently only support 1.10 to 1.12 in this repository.

**Comment by TylerS1066** [comment]
For those of you who have not seen the [7.0.0 beta 9 update](https://www.spigotmc.org/resources/movecraft.31321/update?update=391357) and filled out [our interest form](https://forms.gle/8fUV1bMF34bfXUS99) already, please do!

We have started investigating 1.13+ support and would appreciate everyone filling out the form.  We mainly are looking to find out if we need to support 1.13 or if we can jump straight from 1.10.2-1.12.2 to 1.14+ support (due to some limitations in the 1.13 spigot API).

**Comment by cccm5** [comment]
We've decided to forgo 1.13 specifically, but otherwise the current development version now supports 1.14+. It's still in beta, but regardless this should now be solved.

**Comment by TylerS1066** [comment]
Technically we only support 1.14.4, 1.15.2, and 1.16.5 currently due to the NMS differences between minor versions, but anyone not running these versions really should be.
